zetaclient/bitcoin/bitcoin_client.go

```diff
@@ -682,6 +684,20 @@ func (ob *BTCChainClient) GetInboundVoteMessageFromBtcEvent(inTx *BTCInTxEvnet)
 	amount = amount.Mul(amount, big.NewFloat(1e8))
 	amountInt, _ := amount.Int(nil)
 	message := hex.EncodeToString(inTx.MemoBytes)
+
+	// compliance check
+	receiver := ""
+	if parsedAddress, _, err := common.ParseAddressAndData(message); err == nil {
+		receiver = parsedAddress.Hex()
+	}
+	if config.AnyBannedAddress(inTx.FromAddress, receiver) {
+		logMsg := fmt.Sprintf("Banned address detected in token sent to TSS: sender %s receiver %s tx %s chain %d",
+			inTx.FromAddress, receiver, inTx.TxHash, ob.chain.ChainId)
+		ob.logger.WatchInTx.Warn().Msg(logMsg)
+		ob.logger.Compliance.Warn().Msg(logMsg)
+		return nil
+	}
```

Comment:
I think it would be better to put the full logic in a separate function

Reply:
https://github.com/zeta-chain/node/pull/1790/commits/cf72c1bfc036951b792b8d3d7b327fbb7b53cc32